### ISSUE TITLE
Types: property format in connectLogger options can be a function

### ISSUE THREAD

**Issue by bzums** [log4js-node/log4js-node#666]
You can pass a function as `format` in the `options` parameter of `connectLogger`:
https://github.com/log4js-node/log4js-node/blob/master/lib/connect-logger.js#L255

This is not reflected in its type (https://github.com/log4js-node/log4js-node/blob/master/types/log4js.d.ts#L19)

```ts
export function connectLogger(logger: Logger, options: { format?: string; level?: string; nolog?: any; }): any;
``` 

It should rather be:
```ts
export function connectLogger(logger: Logger, options: {
  format?: string | (req: Request, res: Response, formatFn: (pattern: string) => string) => string; 
  level?: string; 
  nolog?: any;
}): any;
``` 

Ideally the types `Request` and `Response` would be imported from `express`, however if this additional dependency should be omitted, `any` is also possible.

**Comment by nomiddlename** [comment]
Fixed by #763

**Comment by nomiddlename** [comment]
Published to npm in version 3.0.3
